src/env/set_test.c

```diff
@@ -0,0 +1,78 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*   tosh - 21 Shell                                      ::::::::            */
+/*                                                      :+:    :+:            */
+/*   By: aholster <aholster@student.codam.nl>          +:+                    */
+/*       ivan-tey <ivan-tey@student.codam.nl>         +#+                     */
+/*       nloomans <nloomans@student.codam.nl>        +#+                      */
+/*                                                 #+#    #+#                 */
+/*   License: GPLv3                                ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <criterion/criterion.h>
+#include "private.h"
+
+Test(env_set, set_env) {
+	char		*arr[] = {"FOO=foo", "BAR=bar", "UGH=ugh", "NAHHH=nahhh", NULL};
+	t_env		env;
+	int			ret;
+	char		*value;
+
+	ft_bzero(&env, sizeof(t_env));
+	ret = env_from_envp(&env, arr);
+	cr_assert_eq(ret, 0);
+
+	ret = env_set(&env, "YEEY", "yeey");
+	cr_assert_eq(ret, 0);
+	value = env_get(&env, "YEEY");
+	cr_assert_str_eq(value, "yeey");
+}
+
+Test(env_set, change_first_env_value) {
+	char		*arr[] = {"FOO=foo", "BAR=bar", "UGH=ugh", "NAHHH=nahhh", NULL};
+	t_env		env;
+	int			ret;
+	char		*value;
+
+	ft_bzero(&env, sizeof(t_env));
+	ret = env_from_envp(&env, arr);
+	cr_assert_eq(ret, 0);
+
+	ret = env_set(&env, "FOO", "yasss");
+	cr_assert_eq(ret, 0);
+	value = env_get(&env, "FOO");
+	cr_assert_str_eq(value, "yasss");
+}
+
+Test(env_set, change_last_env_value) {
+	char		*arr[] = {"FOO=foo", "BAR=bar", "UGH=ugh", "NAHHH=nahhh", NULL};
+	t_env		env;
+	int			ret;
+	char		*value;
+
+	ft_bzero(&env, sizeof(t_env));
+	ret = env_from_envp(&env, arr);
+	cr_assert_eq(ret, 0);
+
+	ret = env_set(&env, "NAHHH", "yasss");
+	cr_assert_eq(ret, 0);
+	value = env_get(&env, "NAHHH");
+	cr_assert_str_eq(value, "yasss");
+}
+
+Test(env_set, change_middle_env_value) {
+	char		*arr[] = {"FOO=foo", "BAR=bar", "UGH=ugh", "NAHHH=nahhh", NULL};
+	t_env		env;
+	int			ret;
+	char		*value;
+
+	ft_bzero(&env, sizeof(t_env));
+	ret = env_from_envp(&env, arr);
+	cr_assert_eq(ret, 0);
+
+	ret = env_set(&env, "UGH", "yasss");
+	cr_assert_eq(ret, 0);
+	value = env_get(&env, "UGH");
+	cr_assert_str_eq(value, "yasss");
+}
```

Comment:
testing suite incomplete, missing test if rest of environment is mangled post-edit.
also missing test to check if env_set on empty environment results in correct behaviour